sos/plugins/navicli.py

```diff
@@ -67,21 +71,35 @@ def get_navicli_SP_info(self, SP_address):
 
     def setup(self):
         self.get_navicli_config()
+        ipaddr_list = self.get_option_as_list("ipaddrs", delimiter = " ")
         CLARiiON_IP_address_list = []
-        CLARiiON_IP_loop = "stay_in"
-        while CLARiiON_IP_loop == "stay_in":
-            try:
-                ans = input("CLARiiON SP IP Address or [Enter] to exit: ")
-            except:
-                return
-            if self.check_ext_prog("navicli -h %s getsptime" % (ans,)):
-                CLARiiON_IP_address_list.append(ans)
+
+        if len(ipaddr_list) == 0:
+            if not self.batch:
+                CLARiiON_IP_address_list = []
+                CLARiiON_IP_loop = "stay_in"
```

Comment:
The coding style in this plugin is horrendous - if we are doing major surgery on it, we should really fix that too (I'd be perfectly fine with essentially a wholesale rewrite - just keeping (roughly) the same data collection).